python_modules/dagster/dagster/_core/definitions/asset_graph_subset.py

```diff
@@ -205,6 +208,14 @@ def from_storage_dict(
         partitions_subsets_by_asset_key: Dict[AssetKey, PartitionsSubset] = {}
         for key, value in serialized_dict["partitions_subsets_by_asset_key"].items():
             asset_key = AssetKey.from_user_string(key)
+
+            if asset_key not in asset_graph.all_asset_keys:
+                raise DagsterAssetBackfillDataLoadError(
```

Comment:
It's weird to have a backfill-specific error in `AssetGraphSubset`, which can be used in places other than backfills.